CMakeLists.txt

```diff
@@ -280,6 +280,11 @@ if (DISABLE_DOWNLOADS)
 else()
   find_package(Armadillo "${ARMADILLO_VERSION}")
   if (NOT ARMADILLO_FOUND)
+    find_package(BLAS QUIET)
+    find_package(LAPACK QUIET)
+    if (NOT BLAS_FOUND AND NOT LAPACK_FOUND)
```

Comment:
@shrit I recommend to also check for the presence of OpenBLAS, as that can satisfy the BLAS dependency (*).  The condition would then be along the lines of  `if (NOT (BLAS_FOUND OR OPENBLAS_FOUND) AND NOT LAPACK_FOUND)`.

Standard (reference) BLAS is not really used by the vast majority of people, because it's slow.  OpenBLAS is pretty much the de-facto baseline these days.

(*) By default a normal installation of OpenBLAS also comes with built-in LAPACK, but there are distros out there that questionably strip out the built-in LAPACK.

Reply:
I think maybe `find_package(BLAS QUIET)` will already find OpenBLAS (well, depending on the CMake version, but recent-enough CMake versions should find OpenBLAS fine), so it may not be necessary to check specifically for `OPENBLAS_FOUND`.  It can't hurt though.

Reply:
I agree with @rcurtin, we have found that `find_package(BLAS QUITE)` should be able to find `blas`, `openblas`, or any other variant of `BLAS`

Reply:
In that case, should we bump the minimum required version of CMake, to ensure OpenBLAS is detected as BLAS ?

Using standard BLAS (as opposed to OpenBLAS) can lead to poor outcomes for non-expert users (ie. they'll say "I have OpenBLAS but mlpack is still slow").

Reply:
That's true; @shrit do you remember what the minimum version of CMake is that finds OpenBLAS?  I don't mind forcing a newer version of CMake, so long as RHEL7 and relatively recent Ubuntu LTS have it available.  RHEL7 has CMake 3.12 (via the EPEL `cmake3` package).

Reply:
I think it is 3.6

Reply:
Sounds good---I think we can bump our minimum version of CMake then to 3.6; do you want to do that in this PR?  You can change the first line of the root `CMakeLists.txt` to enforce that.  I see that Ubuntu 18.04 has 3.10, which is new enough.